pkg/machinery/scaffold.go

```diff
@@ -375,16 +445,29 @@ func (s scaffold) writeFile(f *file.File) error {
 			return nil
 		case file.Error:
 			// By returning an error, the file is not written and the process will fail
-			return fileAlreadyExistsError{f.Path}
+			return FileAlreadyExistsError{f.Path}
 		}
 	}
 
-	writer, err := s.fs.Create(f.Path)
+	// Create the directory if needed
+	if err := s.fs.MkdirAll(filepath.Dir(f.Path), s.dirPerm); err != nil {
+		return CreateDirectoryError{err}
+	}
+
+	// Create or truncate the file
+	writer, err := s.fs.OpenFile(f.Path, createOrUpdate, s.filePerm)
 	if err != nil {
-		return err
+		return CreateFileError{err}
 	}
+	defer func() {
+		if closeErr := writer.Close(); err == nil && closeErr != nil {
+			err = CloseFileError{err}
```

Comment:
This overwrites `err` in case `Write()` fails. Should be a `log.Error(closeErr)` instead.

Reply:
We check that `err == nil`, so we only overwrite in case there was no error.

Reply:
Right, missed that bit. What if `err != nil && closeErr != nil`?

Reply:
`err` is reported as it happened before. Same as it was before.